packages/compass-workspaces/src/stores/workspaces.ts

```diff
@@ -403,6 +429,26 @@ const reducer: Reducer<WorkspacesState> = (
     };
   }
 
+  if (
+    isAction<CollectionSubtabSelectedAction>(
+      action,
+      WorkspacesActions.CollectionSubtabSelected
+    )
+  ) {
+    return {
+      ...state,
+      tabs: state.tabs.map((tab) => {
+        if (tab.id === action.tabId) {
+          return {
+            ...tab,
+            subTab: action.subTab,
+          };
+        }
+        return tab;
+      }),
+    };
```

Comment:
One small thing caught my eye here on another look. You probably want to short circuit if the subtab is already set to the new value, otherwise you are unnecessarily recreate both tab array and the tab state itself when nothing actually changed. Something like:

```
if (tabToChange.subTab === action.subTab) {
  return state;
}
// ... then your logic
```

Reply:
Just added in [b653813](https://github.com/mongodb-js/compass/pull/5568/commits/b65381312154ef6138f69e08f7eae752e0be1c8c)